File: TypeScript/resp/TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
    for (let i = 0; i < nums.length; i++) {
        let num = nums[i];
        let diff = target - num;
        let idx = nums.indexOf(diff);
        if (idx !== -1 && idx !== i) {
            return [i, idx];
        }
    }
    return [];
}
// function twoSum2(nums: number[], target: number): number[] {
//     const seen = new Map()
//     seen[nums[0]] = 0
//     let i = 1
//     while (i < nums.length) {
//         const remainder = target - nums[i]
//         if (remainder in seen) {
//             return [i, seen[remainder]]
//         }
//         seen[nums[i]] = i
//         i++
//     }
//     return []
// };
export { twoSum };

File: Go/uno.com/pkg/twoSum/twoSum.go
package twoSum

func twoSum_slow(nums []int, target int) []int {
    for i, num := range nums {
		for j, num2 := range nums {
			if i != j && num + num2 == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}

func twoSum(nums []int, target int) []int {
	table := map[int]int{}

	for i, num := range nums {
		if _, ok := table[target-num]; ok {
			return []int{table[target-num], i}
		}

		table[num] = i
	}

	return nil
}